docs/windows.md

```diff
@@ -1,142 +1,94 @@
 # Building on Windows
 
-## Install Git For Windows
-
-As Git is used by the Go language to download dependencies, you need
-to install Git, for example from http://git-scm.com/.
-
-You also must make sure that the directory that contains the Git For
-Windows binary is in the Path environment variable. Note that Git For
-Windows has a 'git' binary in a 'Git\bin' directory and another one in
-a 'Git\cmd' directory. You should only put the 'Git\cmd' directory in
-the Path environment variable.
-
 ## Install Go
 
-Please install the Go language as explained on
-https://golang.org/doc/install.
-
-To properly install Go, you will need to set some environment
-variables. We recommend you to set them globally using the Control
-Panel, as explained in the documentation above, so that these
-environment variables are automatically available in all the possible
-environments that you might want to use like Git Bash, Windows's cmd,
-Cygwin's terminal, Windows' PowerShell and so on.
-
-You must make sure that the GOROOT environment variable is set and
-that the %GOROOT%/bin directory is in the Path environment variable.
-
-The GOPATH environment variable should also be set to a directory that
-you have created, and the %GOPATH/bin directory should also be in the
-Path environment variable.
-
-## Download go-ipfs and fix Git authentication
+`go-ipfs` is built on Golang and thus depends on it for all building methods.  
+https://golang.org/doc/install  
+The `GOPATH` environment variable must be set as well.  
+https://golang.org/doc/code.html#GOPATH
 
-Use the following command to download go-ipfs source code:
-
-```
+## Choose the way you want to proceed
+`go-ipfs` utilizes `make` to automate builds and testing, but can be built without it using only `git` and `go`.  
+The `make` portions cover setting up the environment, "minimal" assumes you have a working version of `go` and `git` already.  
+[make→](#make)  
+[minimal→](#minimal)
+
+## Make
+There are currently 2 documented ways of building with `make` on Windows.  
+[MSYS2→](#msys2)  
+[Cygwin→](#cygwin)
```

Comment:
I'm unsure about adding advice here, it feels subjective. While there are technical differences, at the end of the day both provide the posix tools we need to build go-ipfs. I think the major decision comes down to what users already have installed, if it's neither, the differences between them are just going to be the differences between the 2 sections, which is not much. Mostly it just depends on if you like a graphical package manager or a command line one (*for our purposes).

I'll think this over and see if I can come up with something appropriate. Maybe defer to an external resource comparing the 2.

Reply:
> I think the major decision comes down to what users already have installed

Absolutely. But that doesn’t cover a large group of people.

> I'm unsure about adding advice here, it feels subjective.

It’s totally subjective :)

The problem is that you are *starting* the whole tutorial with a choice, and, in particular, one that has nothing to do with the task at hand (building IPFS) and one users may likely not know enough about in order to *even know how to choose.* That can stop people from following the tutorial right at the beginning. (And even for someone enterprising, it adds a lot of extra work: now they have to go research each of these options in order to figure out which to choose.)

It’s OK for it to be subjective (and you can even say so). The goal is to remove the *barrier* of this choice. This could be as simple as “all these options are good; if you don’t already have one of these tools installed or aren’t familiar with them, go with MSYS2 because it makes the process easy” or “go with ‘minimal’ because you don’t have to install anything extra.”